### DISCUSSION TITLE
How to view the `Actor` object and `Activity` stream – are these HTTP endpoints/routes?

### DISCUSSION THREAD

**Discussion by haszari** [Automattic/wordpress-activitypub#480] [Q&A] [answered]
Hi!

I've installed ActivityPub on my site (https://cartoonbeats.com/), and it's working well. I can follow the whole site stream (aka "feed") by following `@cartoonbeats.com@cartoonbeats.com`.

However, I want to see how this works, and understand what routes / endpoints this plugin adds to my site. 

- Does the plugin add http endpoints i.e. routes?
- How is the ActivityPub data served up to clients?
- If I want to see the actual bytes of a response to learn about the protocol, or debug a problem, how do I do that? E.g. something like `curl https://cartoonbeats.com/activitypub/actor/@cartoonbeats.com` (guessing)

It would be useful to have this info in docs somewhere.

Thanks!

**Comment by pfefferle** [comment, marked as answer]
Hi @haszari 👋

The easiest way to debug the plugin, is by adding `/activitypup` at the end of an URL. For example https://cartoonbeats.com/beats-reality-20231004/activitypub/ (for Post-URLs) or https://cartoonbeats.com/author/haszari/activitypub/ for authors. If that is not possible, you can try to call them with the Accept header `application/activity+json`

```
curl "https://cartoonbeats.com/beats-reality-20231004/" \
     -H 'Accept: application/activity+json'
```

This btw. shows that your setup is not perfect atm. you have enabled both, blog user and authors. In this case, all new posts will be published in the name of the author and the blog-user simply boosts that post. But one of your installed plugins seem to redirect the author URLs to the main URL. That means even if you can follow the blog-user, you won't see new posts (because the author is not accessible). (We need to make this clearer in the WordPress backend)

To fix this issue, you should either

1. disable the author profiles. In this case all new posts will be published in the name of the blog author.
2. or you remove the redirect (I would assume you use YoastSEO).

The endpoints that are added, are listed here: https://cartoonbeats.com/wp-json/activitypub/1.0

I hope this helps :)

**Reply by haszari** [reply]
Thanks for the info @pfefferle - super helpful. Will take a look tonight. I will likely disable author actors, I only want a "site" feed.

**Reply by haszari** [reply]
Disabling the author setting fixed things, I can now follow and [see posts show up in Mastodon](https://mastodon.nz/@cartoonbeats.com@cartoonbeats.com/111175341368808758) 💥
